syncer/syncer.go

```diff
@@ -32,11 +33,13 @@ type layerFetcher interface {
 // Configuration is the config params for syncer
 type Configuration struct {
 	SyncInterval time.Duration
-	// the sync process will try validate the current layer if ValidationDelta has elapsed.
+	// the sync process will try to validate the current layer if ValidationDelta has elapsed.
 	ValidationDelta time.Duration
 	AlwaysListen    bool
 }
 
+var errShuttingDown = errors.New("shutting down")
```

Comment:
I don't think you're supposed to do this. I think you should be using `tomb.ErrDying`. This means that if you kill the tomb then it will return no error if all errors returned by goroutines are `tomb.ErrDying`. If at least one other error is returned by a goroutine then you'll get an aggregated error from the tomb which is the first (non `tomb.ErrDying`) error returned.

You could check for this error yourself, but that's not exactly the same - tomb remembers and returns the first error that isn't `tomb.ErrDying`. So if the first error is `errShuttingDown` and then another error is returned - you'll get `errShuttingDown` as the aggregate (not what you probably want).